matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/profile/DefaultProfileService.kt

```diff
@@ -55,64 +53,38 @@ internal class DefaultProfileService @Inject constructor(private val taskExecuto
                                                          private val userStore: UserStore,
                                                          private val fileUploader: FileUploader) : ProfileService {
 
-    override fun getDisplayName(userId: String, matrixCallback: MatrixCallback<Optional<String>>): Cancelable {
+    override suspend fun getDisplayName(userId: String): Optional<String> {
         val params = GetProfileInfoTask.Params(userId)
-        return getProfileInfoTask
-                .configureWith(params) {
-                    this.callback = object : MatrixCallback<JsonDict> {
-                        override fun onSuccess(data: JsonDict) {
-                            val displayName = data[ProfileService.DISPLAY_NAME_KEY] as? String
-                            matrixCallback.onSuccess(Optional.from(displayName))
-                        }
-
-                        override fun onFailure(failure: Throwable) {
-                            matrixCallback.onFailure(failure)
-                        }
-                    }
-                }
-                .executeBy(taskExecutor)
+        val data = getProfileInfoTask.execute(params)
+        val displayName = data[ProfileService.DISPLAY_NAME_KEY] as? String
+        return Optional.from(displayName)
```

Comment:
Can the taskExecutor be removed from the constructor?

Reply:
Not yet. It's used in `refreshThreePids` below.

Reply:
OK, thanks